Add aggregate method to MongoDB

melohagan opened the pull request:
## Description
You can now use aggregations in the MongoDB connector.

There are two modes:
 - JSON
 - Pipeline

In the JSON mode, you simply provide an array of objects that represent each stage, as per the API docs: https://www.mongodb.com/docs/manual/aggregation/#aggregation-pipeline-example

In the Pipeline mode, you are presented with stages that can be added with the UI. This might be more suitable for "low-code users". 

Currently the json and pipeline modes store data separately, but it could be changed to allow mapping between the two.

Addresses: 
- https://github.com/Budibase/budibase/issues/5387

## Screenshots
**JSON mode**
![Screenshot 2022-10-04 at 11 11 52](https://user-images.githubusercontent.com/101575380/193793944-6910b673-2edc-4d5e-8f0f-88dde7b14582.png)

### Dropdown stages with templates:
![mongo](https://user-images.githubusercontent.com/101575380/195383461-77f5c744-ba7e-4d5f-bf13-cd592d66e5ad.gif)

### Swap stages
![mongo](https://user-images.githubusercontent.com/101575380/195412157-79f88910-4872-4e38-9704-ebf98c04dc68.gif)

Rory-Powell commented:
Nice work mel, this is definitely a more low-code solution. I was wondering about the new `Flow` configuration type, where is that set on the new UI? I wasn't sure from the screenshots, is it the `Action Type` select?

I was thinking the term Flow might be a bit of an overloaded, and possibly something that might make more sense is to display `Action Type` as `Query Type` and use `JSON` or `Form` as the values

melohagan commented:
> Nice work mel, this is definitely a more low-code solution. I was wondering about the new `Flow` configuration type, where is that set on the new UI? I wasn't sure from the screenshots, is it the `Action Type` select?
> 
> I was thinking the term Flow might be a bit of an overloaded, and possibly something that might make more sense is to display `Action Type` as `Query Type` and use `JSON` or `Form` as the values

Changed 'flow' to 'pipeline' and renamed "Action Type" to "Query Type"

shogunpurple reviewed (approved):
LGTM! Great work Mel

melohagan commented on packages/builder/src/components/integration/index.svelte:
```diff
@@ -24,6 +35,37 @@
+  }
+
+  function setEditorTemplate(fromKey, toKey, index) {
+    const currentValue = query.fields.steps[index].value.value
```
Added optional chaining to be sure!
